src/phoenix/server/api/helpers/prompts/models.py

```diff
@@ -1,44 +1,70 @@
 from enum import Enum
-from typing import Any, Union
+from typing import Any, Literal, TypeVar, Union
 
-import strawberry
-from pydantic import BaseModel
+from pydantic import BaseModel, ConfigDict, Field
+from typing_extensions import TypeAlias
 
 JSONSerializable = Union[None, bool, int, float, str, dict[str, Any], list[Any]]
+PromptModel = TypeVar("PromptModel", bound="BasePromptModel")
+
+
+class PromptTemplateType(str, Enum):
+    STRING = "str"
+    CHAT = "chat"
 
 
-@strawberry.enum
 class PromptMessageRole(str, Enum):
     USER = "user"
     SYSTEM = "system"  # e.g. the OpenAI developer role or an Anthropic system instruction
     AI = "ai"  # E.g. the assistant. Normalize to AI for consistency.
     TOOL = "tool"
 
 
-class TextPromptMessage(BaseModel):
+class PromptTemplateFormat(str, Enum):
+    MUSTACHE = "mustache"
+    FSTRING = "fstring"
+    NONE = "none"
+
+
+class BasePromptModel(BaseModel):
```

Comment:
This is too verbose of a name perhaps? I prefer `PhoenixPrompt(BaseModel)` so we aren't using the words "base" and "model" twice without any additional information really